com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderUtil.java

```diff
@@ -29,43 +31,150 @@
-                case "vec4"        : return ShaderDesc.ShaderDataType.SHADER_TYPE_VEC4;
-                case "mat2"        : return ShaderDesc.ShaderDataType.SHADER_TYPE_MAT2;
-                case "mat3"        : return ShaderDesc.ShaderDataType.SHADER_TYPE_MAT3;
-                case "mat4"        : return ShaderDesc.ShaderDataType.SHADER_TYPE_MAT4;
-                case "sampler2D"   : return ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D;
-                case "sampler3D"   : return ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER3D;
-                case "samplerCube" : return ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER_CUBE;
-                default: break;
+            return data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER_CUBE    ||
+                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D       ||
+                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D_ARRAY ||
+                   data_type == ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER3D;
+        }
+
+        private static class ShaderDataTypeConversionEntry {
+            public String                    asStr;
+            public ShaderDesc.ShaderDataType asDataType;
+            public ShaderDataTypeConversionEntry(String str, ShaderDesc.ShaderDataType dataType) {
+                this.asStr      = str;
+                this.asDataType = dataType;
             }
+        }
 
+        private static final ArrayList<ShaderDataTypeConversionEntry> shaderDataTypeConversionLut = new ArrayList<>(Arrays.asList(
+                new ShaderDataTypeConversionEntry("int", ShaderDesc.ShaderDataType.SHADER_TYPE_INT),
+                new ShaderDataTypeConversionEntry("uint", ShaderDesc.ShaderDataType.SHADER_TYPE_UINT),
+                new ShaderDataTypeConversionEntry("float", ShaderDesc.ShaderDataType.SHADER_TYPE_FLOAT),
+                new ShaderDataTypeConversionEntry("vec2", ShaderDesc.ShaderDataType.SHADER_TYPE_VEC2),
+                new ShaderDataTypeConversionEntry("vec3", ShaderDesc.ShaderDataType.SHADER_TYPE_VEC3),
+                new ShaderDataTypeConversionEntry("vec4", ShaderDesc.ShaderDataType.SHADER_TYPE_VEC4),
+                new ShaderDataTypeConversionEntry("mat2", ShaderDesc.ShaderDataType.SHADER_TYPE_MAT2),
+                new ShaderDataTypeConversionEntry("mat3", ShaderDesc.ShaderDataType.SHADER_TYPE_MAT3),
+                new ShaderDataTypeConversionEntry("mat4", ShaderDesc.ShaderDataType.SHADER_TYPE_MAT4),
+                new ShaderDataTypeConversionEntry("sampler2D", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D),
+                new ShaderDataTypeConversionEntry("sampler3D", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER3D),
+                new ShaderDataTypeConversionEntry("samplerCube", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER_CUBE),
+                new ShaderDataTypeConversionEntry("sampler2DArray", ShaderDesc.ShaderDataType.SHADER_TYPE_SAMPLER2D_ARRAY)
+            ));
```

Comment:
I guess we could use Pair<..> here but I felt like this is easier than using "first" and "second" for the members..